pkg/apis/common/v1/association.go

```diff
@@ -188,11 +192,12 @@ func FormatNameWithID(template string, id string) string {
 
 // AssociationConf holds the association configuration of a referenced resource in an association.
 type AssociationConf struct {
-	AuthSecretName string `json:"authSecretName"`
-	AuthSecretKey  string `json:"authSecretKey"`
-	CACertProvided bool   `json:"caCertProvided"`
-	CASecretName   string `json:"caSecretName"`
-	URL            string `json:"url"`
+	AuthSecretName   string `json:"authSecretName"`
+	AuthSecretKey    string `json:"authSecretKey"`
+	IsServiceAccount bool   `json:"isServiceAccount"`
```

Comment:
I was thinking back and forth whether we need this boolean flag. But I think I agree now that an explicit flag is better than trying to detect the type from the secret content. Especially if that would require us to make the application controllers aware of the secret structure which they are not at the moment. Relying on this out of band context as to where to look in the secret is better also for potential "external" secrets.